Sources/Networking/ETagManager.swift

```diff
@@ -18,16 +18,16 @@ class ETagManager {
 
     static let eTagHeaderName = "X-RevenueCat-ETag"
 
-    private let queue = DispatchQueue(label: "ETagManager")
-    private let userDefaults: UserDefaults
-    private let lock = Lock()
+    private let userDefaults: SynchronizedUserDefaults
 
-    init() {
-        self.userDefaults = UserDefaults(suiteName: ETagManager.suiteName) ?? UserDefaults.standard
+    convenience init() {
+        self.init(
+            userDefaults: UserDefaults(suiteName: ETagManager.suiteName) ?? UserDefaults.standard
+        )
```

Comment:
Yeah, possibly. I don’t think there’d be much of an advantage other than potentially making it slightly easier to migrate out of userDefaults should we want to in the future. 
That said, updating it would mean also having some code to migrate the data out of its previous storage so we don’t leave extra data around, and that might not be worth the trouble since code is pretty simple right now

Reply:
The refactor I'm working on will make the data in `ETagManager` incompatible (planning to store the `Data` directly instead of going back and forth from JSON), so I was already thinking of cleaning up the old data anyway.

I think having `ETagManager` handle its own storage makes sense though. We're not currently overriding the `UserDefaults` outside of `ETagManager` so maybe we should remove the parameter?